xapian-core/api/registry.cc

```diff
@@ -197,7 +201,7 @@ Registry::Internal::add_defaults()
     weighting_scheme_short = new Xapian::PL2Weight;
     wtschemes_short["pl2"] = weighting_scheme_short;
     weighting_scheme_short = new Xapian::PL2PlusWeight;
-    wtschemes_short["pl2plus"] = weighting_scheme;
+    wtschemes_short["pl2plus"] = weighting_scheme_short;
```

Comment:
This hard-codes the short names of the weighting schemes, which makes the whole registry approach pointless, as the user can't register their own scheme here.  Instead it should register based on calling a method on the object to get its short name.

Also, we should be able to store the same object in the short map that we already created for the standard map, rather than creating two of each.